File: 215-kth-largest-element-in-an-array/solution.ts
function findKthLargest(nums: number[], k: number): number {
  for (let i = 1; i <= 1e4; i *= 10) {
    const buckets = Array.from({ length: 10 }).map(() => [] as number[]);
    for (const num of nums) {
      const n = Math.floor(((num + 1e4) % (i * 10)) / i);
      buckets[n].push(num);
    }
    nums = buckets.flat();
  }
  return nums[nums.length - k];
}

File: 215-kth-largest-element-in-an-array/solution.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        vector<int> arr(2e4 + 7, 0);
        for(auto &x : nums) arr[x + 1e4]++;
        int now = 0;
        for(int i = 2e4 + 6; i >= 0; i--) {
            now += arr[i];
            if(now >= k) return i - 1e4;
        }
        return -1e4;
    }
};
